tests/Tests_RangeEnhancedGenerator.py

```diff
@@ -0,0 +1,48 @@
+
+# coding: utf-8
+
+# In[ ]:
+
+
+import unittest
+import RangeEnhancedIterator_Generator
+from RangeEnhancedIterator_Generator import GeneratorEnhancedRange
+
+
+class GeneratorTests(unittest.TestCase):
+
+    def setUp(self):
+        print ('Setup called')
+    
+    def test_on_range_w_one_arg(self):
+        example = GeneratorEnhancedRange(2)
+        s = []
+        for i in range(10):
+            for i in example():
+                s.append(i)
+        self.assertEqual(s, [0, 1])
```

Comment:
Do these tests cover all possible failure cases?

Reply:
To be honest, I have no idea. They cover all the cases of the correct usage of the generator. I don't have much knowledge or experience to predict all possible failure cases.